### PR TITLE
Make scan limit error more friendly.

### PR DESCRIPTION
**Pull request by Shylock-Hg**
#### What type of PR is this?
- [ ] bug
- [ ] feature
- [x] enhancement

#### Which issue(s)/PR(s) this PR relates to?
Close https://github.com/vesoft-inc/nebula/issues/3492
  
#### Checklist:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the corresponding label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

#### Release notes:

Please confirm whether to be reflected in release notes and how to describe:
>                                                                 `

### CODE REVIEW

**Comment by czpmango** [comment]
Why do we need this limitation on vertex/edge scans when we already have slow query management?

File: tests/tck/features/match/Base.feature [hunk 1]
```diff
@@ -596,36 +596,36 @@ Feature: Basic match
...
       MATCH (v:player:bachelor) RETURN v
       """
-    Then a ExecutionError should be raised at runtime: Scan vertices must specify limit number.
+    Then a ExecutionError should be raised at runtime: Scan vertices or edges need to specify a limit number, or limit number can not push down.
     When executing query:
       """
       MATCH (v:player{age:23}:bachelor) RETURN v
       """
-    Then a ExecutionError should be raised at runtime: Scan vertices must specify limit number.
+    Then a ExecutionError should be raised at runtime: Scan vertices or edges need to specify a limit number, or limit number can not push down.
```
Comment (czpmango) [inline tests/tck/features/match/Base.feature:614]: These error messages are really not friendly and are implementation-specific.

**Comment by wey-gu** [comment]
It seems this scan limit error now overrides the index not existed case, which is not expected if I understand it correctly.

```
(root@nebula) [amundsen]> MATCH (table_node:Table) return count(table_node) as number_of_tables
[ERROR (-1005)]: Scan vertices must specify limit number.

Sat, 25 Dec 2021 19:30:03 CST

(root@nebula) [amundsen]> CREATE TAG INDEX Table_index on Table();
Execution succeeded (time spent 2003/26690 us)

Sat, 25 Dec 2021 19:32:36 CST

(root@nebula) [amundsen]> REBUILD TAG INDEX Table_index
+------------+
| New Job Id |
+------------+
| 4          |
+------------+
Got 1 rows (time spent 3182/18671 us)

Sat, 25 Dec 2021 19:32:59 CST

(root@nebula) [amundsen]> MATCH (table_node:Table) return count(table_node) as number_of_tables
+------------------+
| number_of_tables |
+------------------+
| 6                |
+------------------+
```

**Comment by wey-gu** [comment]
[ ] - graphD crashed when id(v) IN ["foo"]

```


(root@nebula) [amundsen]> """
                       -> MATCH (db:Database)-[:CLUSTER]->(clstr:Cluster)-[:SCHEMA]->(schema:Schema)-[:TABLE]->(tbl:Table)
                       ->         WHERE id(tbl) == 'hive://gold.test_schema/test_table1'
                       ->         WITH db.name as database_name, clstr.name as cluster_name, schema.name as schema_name, tbl
                       ->         OPTIONAL MATCH (tbl)-[:DESCRIPTION]->(dscrpt:Description)
                       ->         RETURN database_name, cluster_name, schema_name, tbl.name as table_name,
                       ->         dscrpt.description as table_description;
                       -> """
+---------------+--------------+---------------+---------------+-------------------+
| database_name | cluster_name | schema_name   | table_name    | table_description |
+---------------+--------------+---------------+---------------+-------------------+
| "hive"        | "gold"       | "test_schema" | "test_table1" | "1st test table"  |
+---------------+--------------+---------------+---------------+-------------------+
Got 1 rows (time spent 11845/90694 us)

Sat, 25 Dec 2021 19:57:34 CST

(root@nebula) [amundsen]> """
                       -> MATCH (db:Database)-[:CLUSTER]->(clstr:Cluster)-[:SCHEMA]->(schema:Schema)-[:TABLE]->(tbl:Table)
                       ->         WHERE id(tbl) IN ['hive://gold.test_schema/test_table1']
                       ->         WITH db.name as database_name, clstr.name as cluster_name, schema.name as schema_name, tbl
                       ->         OPTIONAL MATCH (tbl:Table)-[:DESCRIPTION]->(dscrpt:Description)
                       ->         RETURN database_name, cluster_name, schema_name, tbl.name as table_name,
                       ->         dscrpt.description as table_description;
                       -> """
2021/12/25 19:56:39 [ERROR] Failed to reconnect, failed to open transport, error: dial tcp 192.168.8.128:9669: connect: connection refused
2021/12/25 19:56:39 Loop error, failed to open transport, error: dial tcp 192.168.8.128:9669: connect: connection refused

Bye root!
Sat, 25 Dec 2021 19:56:39 CST

panic: Loop error, failed to open transport, error: dial tcp 192.168.8.128:9669: connect: connection refused

goroutine 1 [running]:
log.Panicf({0x10500d752, 0xe}, {0x14000291e28, 0x1, 0x1})
	/opt/homebrew/Cellar/go/1.17.5/libexec/src/log/log.go:361 +0xa8
main.main()
	/Users/weyl/dev/nebula-console/main.go:473 +0xcb4
	
```

**Comment by Shylock-Hg** [comment]
> Why do we need this limitation on vertex/edge scans when we already have slow query management?

According to @CPWstatic 's suggestion, avoid user's mistake.

**Review by yixinglu — DISMISSED** [review]

**Review by czpmango — APPROVED** [review]

**Review by yixinglu — APPROVED** [review]

**Review by CPWstatic — APPROVED** [review]

**Pull request merged**